api/server/router/build/build_routes.go

```diff
@@ -67,6 +69,10 @@ func newImageBuildOptions(ctx context.Context, r *http.Request) (*types.ImageBui
 		options.Isolation = i
 	}
 
+	if runtime.GOOS != "windows" && options.SecurityOpt != nil {
+		return nil, fmt.Errorf("the daemon on this platform does not support --security-opt to build")
```

Comment:
Should be BadRequest or Conflict error.
And probably skip the flag and just use `security opts` since this is the daemon?